src/java.base/unix/native/jspawnhelper/jspawnhelper.c

```diff
@@ -146,31 +147,29 @@ int main(int argc, char *argv[]) {
     /* argv[1] contains the fd number to read all the child info */
     int r, fdinr, fdinw, fdout;
     sigset_t unblock_signals;
+    const char* jvm_version;
 
 #ifdef DEBUG
     jtregSimulateCrash(0, 4);
 #endif
 
     if (argc != 3) {
         fprintf(stdout, "Incorrect number of arguments: %d\n", argc);
-        shutItDown();
+        shutItDown("Unknown");
     }
 
-    if (strcmp(argv[1], VERSION_STRING) != 0) {
-        fprintf(stdout, "Incorrect Java version: %s\n", argv[1]);
```

Comment:
Don't you want to keep the comparison and print the warning if the versions don't match?

```
if (strcmp(argv[1], VERSION_STRING) != 0) {
    fprintf(stdout, "Warning: Version mismatch: %s != %s\n", argv[1], VERSION_STRING);
}
```

Reply:
I understand this mismatch is otherwise only printed in `shutItDown`

Reply:
I was not able to convince myself that printing into `stdout` from jspawnhelper does not introduce more problems. So in the interest of extra safety, I only did the version printing from the diagnostic paths where we already print.